multicluster/controllers/multicluster/leader/clusterinfo_export_handler.go

```diff
@@ -41,7 +41,7 @@ func (r *ResourceExportReconciler) handleClusterInfo(ctx context.Context, req ct
 	}
 
 	if !resExport.DeletionTimestamp.IsZero() {
-		if common.StringExistsInSlice(resExport.Finalizers, constants.ResourceExportFinalizer) {
+		if common.AnyExistsInSlice(resExport.Finalizers, []string{constants.LegacyResourceExportFinalizer, constants.ResourceExportFinalizer}) {
```

Comment:
I think we could just use `slices.Contains(...) || slices.Contains()` and avoid introducing the `AnyExistsInSlice` since we only have 2 elements to check, but that's not too important.